core/geometry/src/polyface/PolyfaceBuilder.ts

```diff
@@ -2133,3 +2138,50 @@ function resolveToIndexedXYZCollectionOrCarrier(points: Point3d[] | LineString3d
 function distinctIndices(i0: number, i1: number, i2: number): boolean {
   return i0 !== i1 && i1 !== i2 && i2 !== i0;
 }
+
+/**
+ * In this file to avoid topology -> polyface dependency.
+ * @internal
+ */
+export namespace HalfEdgeGraphSearch {
+  /**
+   * Search the graph for an interior face containing the test point(s).
+   * * For best results:
+   *   * the graph exterior face(s) should be masked with HalfEdgeMask.EXTERIOR
+   *   * the graph should contain no vertical faces
+   * @param graph input topology, z-coordinates ignored
+   * @param testPoint xy point(s) to search
+   * @returns edge in the face loop of the containing face, or `undefined` if `testPoint` is in an exterior face,
+   * or an array of the same in 1-1 correspondence with the input points.
+   */
+  export function findContainingFaceXY(graph: HalfEdgeGraph, testPoint: XAndY | XAndY[]): (HalfEdge | undefined)[] | HalfEdge | undefined {
+    const results: (HalfEdge | undefined)[] = [];
+    const idToFaceIndexMap = graph.constructIdToFaceIndexMap();
+    const facetIndexToFaceIndexMap: number[] = [];
+    const acceptFace = (face: HalfEdge): boolean => {
```

Comment:
This is a bit confusing. `idToFaceIndexMap` stores node's (half-edge) id as key and min index as value. Now `acceptFace` takes each node (not each face) and pushes a number into `facetIndexToFaceIndexMap`

First of all I think you should rename `face` to `node`. Also why a number array is called Map? (`facetIndexToFaceIndexMap`) It's not a key/value map. It's just an array.

Finally, I'm confused about difference between face and facet here!

Reply:
A `HalfEdge` can represent a vertex, edge, or face depending on context. So calling it a "vertex", "edge", or "face" gives a clue as to this context, and thus I usually prefer that over the generic "node". In this light, we have for `acceptFace`:

* it is called once per graph face
* the mask it checks is a face mask
* the index it stores in the array is a face index

And these three facts suggest the name of its input should be "face".

The array used to be a Map before I realized its keys are sequential and thus it is more efficient as an array. I'll rename.

A "face" is a part of a graph or a solid (brep).  A "facet" is part of a mesh (`Polyface`).